content/tutorial-air-quality-analysis.md

```diff
@@ -0,0 +1,360 @@
+                        vcompute_indices('SO2', pollutants[..., 4]),
+                        vcompute_indices('CO', pollutants[..., 5]),
+                        vcompute_indices('O3', pollutants[..., 6])), axis=1)
+```
+
+### Air quality indices
+
+Using [np.amax](https://numpy.org/devdocs/reference/generated/numpy.amax.html), we find out the maximum sub-index for each period, which is our Air Quality Index!
+
+```{code-cell} ipython3
+aqi_array = np.amax(sub_indices, axis=1)
+```
+
+With this, we have the AQI for every hour from June 1, 2019 to June 30, 2020. Note that even though we started out with
+the data from 31st May, we truncated that during the moving averages step.
+
++++
+
+## Paired Student's t-test on the AQIs
+
+Hypothesis testing is a form of descriptive statistics used to help us make decisions with the data. From the calculated AQI data, we want to find out if there was a statistically significant difference in average AQI before and after the lockdown was imposed. We will use the left-tailed, [paired Student's t-test](https://en.wikipedia.org/wiki/Student%27s_t-test#Dependent_t-test_for_paired_samples) to compute two test statistics- the [`t statistic`](https://en.wikipedia.org/wiki/T-statistic) and the [`p value`](https://en.wikipedia.org/wiki/P-value). We will then compare these with the corresponding critical values to make a decision.
+
+![Normal distribution plot showing area of rejection in one-tailed test (left tailed)](_static/11-one-tailed-test.svg)
+
+### Sampling
+
+We will now import the `datetime` column from our original dataset into a [*datetime64* dtype](https://numpy.org/devdocs/reference/arrays.scalars.html) array. We will use this array to index the AQI array and obtain subsets of the dataset.
+
+```{code-cell} ipython3
+datetime = np.genfromtxt("air-quality-data.csv", dtype='M8[h]', delimiter=",",
+                         skip_header=1, usecols=(0, ))[-(pollutants_A_24hr_avg.shape[0]):]
+```
+
+Since total lockdown commenced in Delhi from March 24, 2020, the after-lockdown subset is of the period March 24, 2020 to June 30, 2020. The before-lockdown subset is for the same length of time before 24th March.
+
+```{code-cell} ipython3
+after_lock = aqi_array[np.where(datetime >= np.datetime64('2020-03-24T00'))]
+
+before_lock = aqi_array[np.where(datetime <= np.datetime64('2020-03-21T00'))][-(after_lock.shape[0]):]
+
+print(after_lock.shape)
+print(before_lock.shape)
+```
+
+To make sure our samples are *approximately* normally distributed, we take samples of size `n = 30`. `before_sample_A` and `after_sample_A` are the set of random observations drawn before and after the total lockdown. We use [np.random.permutation](https://numpy.org/devdocs/reference/random/generated/numpy.random.permutation.html) to generate the samples.
+
+```{code-cell} ipython3
+before_sample = np.random.permutation(before_lock)[:30]
+after_sample = np.random.permutation(after_lock)[:30]
```

Comment:
We should be using the Generator-based random interface here, since it is the recommended interface:
```suggestion
rng = np.random.default_rng()
before_sample = rng.permutation(before_lock)[:30]
after_sample = rng.permutation(after_lock)[:30]
```

Also, since this is sampling from a dataset, you may want to consider `choice` instead, though your way may be faster :)
```python
before_sample = rng.choice(before_lock, size=30, replace=False)
```

Reply:
`choice` is in fact faster so I have decided to use it!